compiler/ast/ast.go

```diff
@@ -179,21 +274,30 @@ type VectorValue struct {
 
 type MapExpr struct {
 	Kind    string      `json:"kind" unpack:""`
+	Lpipe   int         `json:"lpipe"`
 	Entries []EntryExpr `json:"entries"`
+	Rpipe   int         `json:"rpipe"`
 }
 
+func (m *MapExpr) Pos() int { return m.Lpipe }
+func (m *MapExpr) End() int { return m.Rpipe + 1 }
+
 type EntryExpr struct {
 	Key   Expr `json:"key"`
 	Value Expr `json:"value"`
 }
 
 type OverExpr struct {
 	Kind   string `json:"kind" unpack:""`
+	Left   int    `json:"left"`
```

Comment:
If I understand correctly, this is the position of the "over" keyword, so maybe call this KeywordPos?

Reply:
For sure, not sure why I did this